services/nodeping/nodeping-uptime.service.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const { BaseJsonService } = require('..')
+const Joi = require('joi')
+
+const rowSchema = Joi.object().keys({
+  uptime: Joi.number()
+    .precision(3)
+    .min(0)
+    .max(100),
+})
+
+const schema = Joi.array()
+  .items(rowSchema)
+  .min(1)
+
+/*
+ * this is the checkUuid for the NodePing.com (as used on the [example page](https://nodeping.com/reporting.html#results))
+ */
+const sampleCheckUuid = 'jkiwn052-ntpp-4lbb-8d45-ihew6d9ucoei'
+
+// TODO: support for custom # of days
+// TODO: support for custom color thresholds
+// TODO: support for custom colors
+// TODO: support for custom '100%' label
+// TODO: support for custom # of decimal places
+
+class NodePingUptime extends BaseJsonService {
+  static get category() {
+    return 'monitoring'
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'Uptime',
+    }
+  }
+
+  static get route() {
+    return {
+      base: 'nodeping/uptime',
+      pattern: ':checkUuid',
+      queryParams: ['label'],
```

Comment:
You don't need to re-define the `label` query param here, this is default option available on all badges